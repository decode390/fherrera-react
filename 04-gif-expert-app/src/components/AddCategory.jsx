import { useState } from 'react'
import { PropTypes } from "prop-types";


export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({target})=>{
    setInputValue(target.value)
  }

  const onSubmitForm = (evt)=>{
    evt.preventDefault();
    if (inputValue.trim().length <= 1) return;

    onNewCategory(inputValue.trim());
    /* setCategories(categories=>[inputValue, ...categories]); */
    setInputValue('');
  }

  return (
    <form aria-label='search' onSubmit={onSubmitForm}>
      <input 
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}
