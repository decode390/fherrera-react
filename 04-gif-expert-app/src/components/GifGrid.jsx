import { useFetchGifs } from '../hooks/useFetchGifs';
import {GiftItem} from '.';
import { PropTypes } from "prop-types";

export const GifGrid = ({category}) => {
  const {images, isLoading} = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <div className='card-grid'>
        {
          isLoading 
          ? (<h2>Cargando...</h2>)
          : images.map((image) => 
            <GiftItem 
              key={image.id} 
              {...image}
            />
          )
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
