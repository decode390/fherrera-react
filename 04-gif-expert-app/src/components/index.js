export * from './GifGrid';
export * from './AddCategory';
export * from './GiftItem';
