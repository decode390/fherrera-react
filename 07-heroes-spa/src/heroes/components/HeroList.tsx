import { ValidPublishers, getHeroesByPublisher } from '@heroes/helpers';
import { HeroCard } from './HeroCard';
import { useMemo } from 'react';

export const HeroList = ({publisher}: {publisher: ValidPublishers}) => {
  const heroes = useMemo(() => getHeroesByPublisher(publisher), [publisher]);

  return (
    <>
      <div className="row rows-col-1 row-cols-md-3 g-3">
        {
          heroes.map(
            hero => <HeroCard key={hero.id} hero={hero}/>
          )
        }
      </div>
    </>
  )
}
