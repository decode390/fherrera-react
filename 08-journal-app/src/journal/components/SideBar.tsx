import { TurnedInNot } from '@mui/icons-material';
import { 
  Box, 
  Divider, 
  Drawer, 
  Grid, 
  List, 
  ListItem, 
  ListItemButton, 
  ListItemIcon, 
  ListItemText, 
  Toolbar, 
  Typography 
} from '@mui/material';

export const SideBar = ({ drawerWidth = 240 }: SideBarProps) => {
  return (
    <Box 
      component='nav'
      sx={{ width: { sm: `${drawerWidth}px` }, flexShrink: { sm: 0 } }}
    >
      <Drawer
        variant="permanent"
        open
        sx={{
          display: { xs: 'block' },
          '& .MuiDrawer-paper': { 
            boxSizing: 'border-box',
            width: `${drawerWidth}px` 
          }
        }}
      >

        <Toolbar>
          <Typography
            variant="h6"
            noWrap
            component='div'
          >
            John Doe
          </Typography>
        </Toolbar>

        <Divider/>

        <List>
          {
            ['Enero', 'Febrero', 'Marzo', 'Abril'].map(text => (
              <ListItem key={text} disablePadding>

                <ListItemButton>

                  <ListItemIcon>
                    <TurnedInNot/>
                  </ListItemIcon>

                  <Grid container>
                    <ListItemText primary={text}/>
                    <ListItemText secondary={ 'Lorem ipsum dolor sit amet, qui minim labore' }/>
                  </Grid>

                </ListItemButton>

              </ListItem>
            ))
          }
        </List>

      </Drawer>

    </Box>
  )
}


interface SideBarProps {
  drawerWidth?: Number
}
