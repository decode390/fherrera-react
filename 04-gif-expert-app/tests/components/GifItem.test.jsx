import { render, screen } from "@testing-library/react"
import { GiftItem } from "../../src/components/GiftItem"

describe('test in GifItem', () => {
  const title = 'TitleTest';
  const url = 'http://test.img.com/img.jpg';

  test('should be match with snapshot', () => {
    const {container} = render(
      <GiftItem 
        title={title}
        url={url}
      />
    );
    /* screen.debug() */
    expect(container).toMatchSnapshot();
  });

  test('should be load src and alt in img', () => {
    render(<GiftItem title={title} url={url}/>);
    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should be show title', () => {
    render(<GiftItem title={title} url={url}/>);
    expect(screen.getByText(title)).toBeTruthy();
  })
});
