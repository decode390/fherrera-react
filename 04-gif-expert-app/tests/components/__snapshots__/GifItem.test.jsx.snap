// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifItem should be match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="TitleTest"
      src="http://test.img.com/img.jpg"
    />
    <p>
      TitleTest
    </p>
  </div>
</div>
`;
